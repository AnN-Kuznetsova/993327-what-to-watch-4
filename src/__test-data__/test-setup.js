import Adapter from "enzyme-adapter-react-16";
import Enzyme from "enzyme";

Enzyme.configure({
  adapter: new Adapter(),
});

const noop = () => {};
Object.defineProperty(window, `scrollTo`, {value: noop, writable: true});
Object.defineProperty(HTMLMediaElement.prototype, `play`, {value: noop, writable: true});
Object.defineProperty(HTMLMediaElement.prototype, `pause`, {value: noop, writable: true});
Object.defineProperty(HTMLMediaElement.prototype, `load`, {value: noop, writable: true});
