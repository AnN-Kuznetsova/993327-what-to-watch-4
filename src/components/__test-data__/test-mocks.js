const promoMovie = {
  title: `The Grand Budapest Hotel`,
  genre: `Drama`,
  releaseDate: new Date(2014, 0),
  posterUrl: `img/the-grand-budapest-hotel-poster.jpg`,
  backgroundUrl: `img/bg-the-grand-budapest-hotel.jpg`,
};

const movieTitles = [
  `Fantastic Beasts: The Crimes of Grindelwald`,
  `Bohemian Rhapsody`,
  `Macbeth`,
  `Aviator`,
  `We need to talk about Kevin`,
  `What We Do in the Shadows`,
  `Revenant`,
  `Johnny English`,
  `Shutter Island`,
  `Pulp Fiction`,
  `No Country for Old Men`,
  `Snatch`,
  `Moonrise Kingdom`,
  `Seven Years in Tibet`,
  `Midnight Special`,
  `War of the Worlds`,
  `Dardjeeling Limited`,
  `Orlando`,
  `Mindhunter`,
  `Midnight Special`,
];

const genreName = `Kids & Family`;

const movieTitle = `Aviator`;


export {
  genreName,
  movieTitle,
  movieTitles,
  promoMovie,
};
