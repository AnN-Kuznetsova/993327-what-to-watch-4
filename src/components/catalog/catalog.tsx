
import * as React from "react";
import {connect} from "react-redux";

import {ActionCreator} from "../../reducers/application/application";
import {MoviePropType} from "../../prop-types";
import {SmallMovieCardWithVideoPlayer} from "../small-movie-card/small-movie-card";
import {ShowMoreButton} from "../show-more-button/show-more-button";
import {getMoviesForCatalog} from "../../reducers/data/selectors";
import {getVisibleMoviesCount} from "../../reducers/application/selectors";


const CatalogComponent = (props) => {
  const {
    movies,
    visibleCardCount,
    onShowMoreButtonClick,
  } = props;

  if (movies && movies.length) {
    return (
      <React.Fragment>
        <div className="catalog__movies-list">
          {movies.slice(0, visibleCardCount)
            .map((movie, index) =>
              <SmallMovieCardWithVideoPlayer
                key={movie.title + index}
                movie={movie}
              />
            )}
        </div>

        {
          (movies.length > visibleCardCount) &&
          <ShowMoreButton onClick={onShowMoreButtonClick} />
        }
      </React.Fragment>
    );
  }

  return null;
};


CatalogComponent.propTypes = {
  movies: PropTypes.arrayOf(MoviePropType),
  visibleCardCount: PropTypes.number.isRequired,
  onShowMoreButtonClick: PropTypes.func.isRequired,
};


const mapStateToProps = (state) => ({
  movies: getMoviesForCatalog(state),
  visibleCardCount: getVisibleMoviesCount(state),
});

const mapDispatchToProps = (dispatch) => ({
  onShowMoreButtonClick() {
    dispatch(ActionCreator.incrementVisibleMoviesCount());
  },
});

const Catalog = connect(mapStateToProps, mapDispatchToProps)(CatalogComponent);


export {
  CatalogComponent,
  Catalog,
};
