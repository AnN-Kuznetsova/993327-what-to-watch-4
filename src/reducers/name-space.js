export const NameSpace = {
  DATA: `DATA`,
  APPLICATION: `APPLICATION`,
  USER: `USER`,
};
