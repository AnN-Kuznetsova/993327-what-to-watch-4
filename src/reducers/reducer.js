import {
  NUMBER_OF_CARDS_IN_CATALOG_AT_STARTUP,
  NUMBER_OF_CARDS_TO_INCREMENT,
  PageType,
  FilterType,
} from "../const";
import {
  extend,
  getFilteredMovies,
} from "../utils/utils";

import {movies} from "../mocks/movies";
import {promoMovie} from "../mocks/promo-movie";


const initialState = {
  movies,
  genre: `All genres`,
  movieList: movies,
  visibleMoviesCount: movies.length <= NUMBER_OF_CARDS_IN_CATALOG_AT_STARTUP ? movies.length :
    NUMBER_OF_CARDS_IN_CATALOG_AT_STARTUP,
  activeMovie: promoMovie,
  activePage: PageType.MAIN,
  prevPage: PageType.MAIN,
};


const ActionType = {
  CHANGE_GENRE: `CHANGE_GENRE`,
  GET_MOVIES: `GET_MOVIES`,
  CHANGE_ACTIVE_MOVIE: `CHANGE_ACTIVE_MOVIE`,
  CHANGE_ACTIVE_PAGE: `CHANGE_ACTIVE_PAGE`,
  INCREMENT_VISIBLE_MOVIES_COUNT: `INCREMENT_VISIBLE_MOVIES_COUNT`,
  RESET_VISIBLE_MOVIES_COUNT: `RESET_VISIBLE_MOVIES_COUNT`,
};

const ActionCreator = {
  changeGenre: (value) => ({
    type: ActionType.CHANGE_GENRE,
    payload: value,
  }),
  getMovies: (moviesCount = null) => ({
    type: ActionType.GET_MOVIES,
    payload: moviesCount,
  }),
  changeActiveMovie: (movie) => ({
    type: ActionType.CHANGE_ACTIVE_MOVIE,
    payload: movie,
  }),
  changeActivePage: (page) => ({
    type: ActionType.CHANGE_ACTIVE_PAGE,
    payload: page,
  }),
  incrementVisibleMoviesCount: () => ({
    type: ActionType.INCREMENT_VISIBLE_MOVIES_COUNT,
    payload: NUMBER_OF_CARDS_TO_INCREMENT,
  }),
  resetVisibleMoviesCount: () => ({
    type: ActionType.RESET_VISIBLE_MOVIES_COUNT,
    payload: null,
  }),
};


const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.CHANGE_GENRE:
      return extend(state, {
        genre: action.payload,
      });

    case ActionType.GET_MOVIES:
      return extend(state, {
        movieList: getFilteredMovies(state.movies, FilterType.GENRE, state.genre, state.activeMovie, action.payload),
      });

    case ActionType.CHANGE_ACTIVE_MOVIE:
      return extend(state, {
        activeMovie: action.payload,
      });

    case ActionType.CHANGE_ACTIVE_PAGE:
      if (state.activePage === action.payload) {
        return state;
      }
      return extend(state, {
        prevPage: state.activePage,
        activePage: action.payload,
      });

    case ActionType.INCREMENT_VISIBLE_MOVIES_COUNT:
      return extend(state, {
        visibleMoviesCount: state.visibleMoviesCount + action.payload,
      });

    case ActionType.RESET_VISIBLE_MOVIES_COUNT:
      return extend(state, {
        visibleMoviesCount: initialState.visibleMoviesCount,
      });

    default:
      return state;
  }
};


export {
  ActionCreator,
  ActionType,
  reducer,
};
