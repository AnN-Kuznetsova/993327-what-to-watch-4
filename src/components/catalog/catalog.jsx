import PropTypes from "prop-types";
import React from "react";
import {connect} from "react-redux";

import {NUMBER_OF_CARDS_IN_CATALOG_AT_STARTUP} from "../../const";
import {ActionCreator} from "../../reducers/reducer";
import {MoviePropType} from "../../prop-types";
import {SmallMovieCardWithVideoPlayer} from "../small-movie-card/small-movie-card";
import {ShowMoreButton} from "../show-more-button/show-more-button";
import {withFilter} from "../../hocs/with-filter/with-filter";


const getGenreFilterNames = (movies) => {
  const filterNames = new Set();

  movies.forEach((movie) => {
    movie.genres.forEach((genre) => filterNames.add(genre));
  });

  return Array.from(filterNames);
};


const CatalogComponent = (props) => {
  const {
    movies,
    movieList,
    onSmallMovieCardClick,
    renderFilter,
    activeFilter,
    onFilterClick,
  } = props;

  const handleShowMoreButtonClick = () => {};

  return (
    <React.Fragment>
      {renderFilter && renderFilter(getGenreFilterNames(movies), onFilterClick, activeFilter)}

      <div className="catalog__movies-list">
        {
          movieList.map((movie, index) =>
            <SmallMovieCardWithVideoPlayer
              key={movie.title + index}
              movie={movie}
              onClick={onSmallMovieCardClick}
            />
          )
        }
      </div>

      {
        (movieList.length > NUMBER_OF_CARDS_IN_CATALOG_AT_STARTUP) &&
        <ShowMoreButton onClick={handleShowMoreButtonClick} />
      }
    </React.Fragment>
  );
};


CatalogComponent.propTypes = {
  movies: PropTypes.arrayOf(MoviePropType).isRequired,
  movieList: PropTypes.arrayOf(MoviePropType).isRequired,
  onSmallMovieCardClick: PropTypes.func.isRequired,
  renderFilter: PropTypes.func,
  activeFilter: PropTypes.string.isRequired,
  onFilterClick: PropTypes.func.isRequired,
};


const mapStateToProps = (state) => ({
  activeFilter: state.genre,
  movies: state.movies,
  movieList: state.movieList,
});

const mapDispatchToProps = (dispatch) => ({
  onFilterClick(newFilterValue) {
    dispatch(ActionCreator.changeGenre(newFilterValue));
    dispatch(ActionCreator.getMovies());
  },
});

const Catalog = connect(mapStateToProps, mapDispatchToProps)(CatalogComponent);
const CatalogWithFilterByGenre = withFilter(Catalog);


export {
  CatalogComponent,
  Catalog,
  CatalogWithFilterByGenre,
};
