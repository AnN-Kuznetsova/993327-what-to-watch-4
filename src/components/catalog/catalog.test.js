import React from "react";
import configureStore from "redux-mock-store";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";

import {CatalogComponent} from "./catalog";
import {NameSpace} from "../../reducers/name-space";

import {mockMovies} from "../../__test-data__/test-mocks";


const mockStore = configureStore([]);

const store = mockStore({
  [NameSpace.APPLICATION]: {
    activePage: ``,
    prevPage: ``,
    playerStartTime: 0,
  },
});

const props = {
  movies: [],
  visibleCardCount: 8,
  onSmallMovieCardClick: () => {},
  onShowMoreButtonClick: () => {},
};

const nodeMock = {
  createNodeMock: () => {
    return {};
  }
};


describe(`Render Catalog`, () => {
  it(`Catalog should match with snapshot with ShowMoreButton`, () => {
    props.movies = [].concat(mockMovies, mockMovies, mockMovies);

    const catalogSnapshot = renderer.create(
        <Provider store={store}>
          <CatalogComponent {...props} />
        </Provider>, nodeMock
    ).toJSON();

    expect(catalogSnapshot).toMatchSnapshot();
  });


  it(`Catalog should match with snapshot without ShowMoreButton`, () => {
    props.movies = mockMovies;

    const catalogSnapshot = renderer.create(
        <Provider store={store}>
          <CatalogComponent {...props} />
        </Provider>, nodeMock
    ).toJSON();

    expect(catalogSnapshot).toMatchSnapshot();
  });
});
