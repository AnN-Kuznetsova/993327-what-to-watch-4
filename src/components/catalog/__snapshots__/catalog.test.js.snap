// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Catalog Should match with snapshot when active page is MAIN_INDEX 1`] = `
<section
  className="catalog "
>
  <h2
    className="catalog__title visually-hidden"
  >
    Catalog
  </h2>
  <ul
    className="catalog__genres-list"
  >
    <li
      className="catalog__genres-item"
    >
      <a
        className="catalog__genres-link"
        href="#"
        onClick={[Function]}
      >
        All genres
      </a>
    </li>
    <li
      className="catalog__genres-item"
    >
      <a
        className="catalog__genres-link"
        href="#"
        onClick={[Function]}
      >
        Comedies
      </a>
    </li>
    <li
      className="catalog__genres-item"
    >
      <a
        className="catalog__genres-link"
        href="#"
        onClick={[Function]}
      >
        Crime
      </a>
    </li>
    <li
      className="catalog__genres-item"
    >
      <a
        className="catalog__genres-link"
        href="#"
        onClick={[Function]}
      >
        Documentary
      </a>
    </li>
    <li
      className="catalog__genres-item"
    >
      <a
        className="catalog__genres-link"
        href="#"
        onClick={[Function]}
      >
        Dramas
      </a>
    </li>
    <li
      className="catalog__genres-item"
    >
      <a
        className="catalog__genres-link"
        href="#"
        onClick={[Function]}
      >
        Horror
      </a>
    </li>
    <li
      className="catalog__genres-item"
    >
      <a
        className="catalog__genres-link"
        href="#"
        onClick={[Function]}
      >
        Kids & Family
      </a>
    </li>
    <li
      className="catalog__genres-item"
    >
      <a
        className="catalog__genres-link"
        href="#"
        onClick={[Function]}
      >
        Romance
      </a>
    </li>
    <li
      className="catalog__genres-item"
    >
      <a
        className="catalog__genres-link"
        href="#"
        onClick={[Function]}
      >
        Sci-Fi
      </a>
    </li>
    <li
      className="catalog__genres-item"
    >
      <a
        className="catalog__genres-link"
        href="#"
        onClick={[Function]}
      >
        Thrillers
      </a>
    </li>
  </ul>
  <div
    className="catalog__movies-list"
  >
    <article
      className="small-movie-card catalog__movies-card"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <div
          className=""
        >
          <video
            autoPlay={false}
            className="player__video"
            poster="img/bohemian-rhapsody.jpg"
            style={
              Object {
                "height": 175,
              }
            }
          />
        </div>
        <img
          alt="Bohemian Rhapsody"
          height="175"
          src="img/bohemian-rhapsody.jpg"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page.html"
        >
          Bohemian Rhapsody
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <div
          className=""
        >
          <video
            autoPlay={false}
            className="player__video"
            poster="img/aviator.jpg"
            style={
              Object {
                "height": 175,
              }
            }
          />
        </div>
        <img
          alt="Aviator"
          height="175"
          src="img/aviator.jpg"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page.html"
        >
          Aviator
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <div
          className=""
        >
          <video
            autoPlay={false}
            className="player__video"
            poster="img/revenant.jpg"
            style={
              Object {
                "height": 175,
              }
            }
          />
        </div>
        <img
          alt="Revenant"
          height="175"
          src="img/revenant.jpg"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page.html"
        >
          Revenant
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <div
          className=""
        >
          <video
            autoPlay={false}
            className="player__video"
            poster="img/johnny-english.jpg"
            style={
              Object {
                "height": 175,
              }
            }
          />
        </div>
        <img
          alt="Johnny English"
          height="175"
          src="img/johnny-english.jpg"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page.html"
        >
          Johnny English
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <div
          className=""
        >
          <video
            autoPlay={false}
            className="player__video"
            poster="img/shutter-island.jpg"
            style={
              Object {
                "height": 175,
              }
            }
          />
        </div>
        <img
          alt="Shutter Island"
          height="175"
          src="img/shutter-island.jpg"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page.html"
        >
          Shutter Island
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <div
          className=""
        >
          <video
            autoPlay={false}
            className="player__video"
            poster="img/pulp-fiction.jpg"
            style={
              Object {
                "height": 175,
              }
            }
          />
        </div>
        <img
          alt="Pulp Fiction"
          height="175"
          src="img/pulp-fiction.jpg"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page.html"
        >
          Pulp Fiction
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <div
          className=""
        >
          <video
            autoPlay={false}
            className="player__video"
            poster="img/seven-years-in-tibet.jpg"
            style={
              Object {
                "height": 175,
              }
            }
          />
        </div>
        <img
          alt="Seven Years in Tibet"
          height="175"
          src="img/seven-years-in-tibet.jpg"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page.html"
        >
          Seven Years in Tibet
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <div
          className=""
        >
          <video
            autoPlay={false}
            className="player__video"
            poster="img/snatch.jpg"
            style={
              Object {
                "height": 175,
              }
            }
          />
        </div>
        <img
          alt="Snatch"
          height="175"
          src="img/snatch.jpg"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page.html"
        >
          Snatch
        </a>
      </h3>
    </article>
  </div>
  <div
    className="catalog__more"
  >
    <button
      className="catalog__button"
      onClick={[Function]}
      type="button"
    >
      Show more
    </button>
  </div>
</section>
`;

exports[`Render Catalog Should match with snapshot when active page is MAIN_MOVIE_DETAILS 1`] = `
<section
  className="catalog catalog--like-this"
>
  <h2
    className="catalog__title "
  >
    More like this
  </h2>
  <div
    className="catalog__movies-list"
  >
    <article
      className="small-movie-card catalog__movies-card"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <div
          className=""
        >
          <video
            autoPlay={false}
            className="player__video"
            poster="img/bohemian-rhapsody.jpg"
            style={
              Object {
                "height": 175,
              }
            }
          />
        </div>
        <img
          alt="Bohemian Rhapsody"
          height="175"
          src="img/bohemian-rhapsody.jpg"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page.html"
        >
          Bohemian Rhapsody
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <div
          className=""
        >
          <video
            autoPlay={false}
            className="player__video"
            poster="img/aviator.jpg"
            style={
              Object {
                "height": 175,
              }
            }
          />
        </div>
        <img
          alt="Aviator"
          height="175"
          src="img/aviator.jpg"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page.html"
        >
          Aviator
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <div
          className=""
        >
          <video
            autoPlay={false}
            className="player__video"
            poster="img/revenant.jpg"
            style={
              Object {
                "height": 175,
              }
            }
          />
        </div>
        <img
          alt="Revenant"
          height="175"
          src="img/revenant.jpg"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page.html"
        >
          Revenant
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <div
          className=""
        >
          <video
            autoPlay={false}
            className="player__video"
            poster="img/johnny-english.jpg"
            style={
              Object {
                "height": 175,
              }
            }
          />
        </div>
        <img
          alt="Johnny English"
          height="175"
          src="img/johnny-english.jpg"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page.html"
        >
          Johnny English
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <div
          className=""
        >
          <video
            autoPlay={false}
            className="player__video"
            poster="img/shutter-island.jpg"
            style={
              Object {
                "height": 175,
              }
            }
          />
        </div>
        <img
          alt="Shutter Island"
          height="175"
          src="img/shutter-island.jpg"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page.html"
        >
          Shutter Island
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <div
          className=""
        >
          <video
            autoPlay={false}
            className="player__video"
            poster="img/pulp-fiction.jpg"
            style={
              Object {
                "height": 175,
              }
            }
          />
        </div>
        <img
          alt="Pulp Fiction"
          height="175"
          src="img/pulp-fiction.jpg"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page.html"
        >
          Pulp Fiction
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <div
          className=""
        >
          <video
            autoPlay={false}
            className="player__video"
            poster="img/seven-years-in-tibet.jpg"
            style={
              Object {
                "height": 175,
              }
            }
          />
        </div>
        <img
          alt="Seven Years in Tibet"
          height="175"
          src="img/seven-years-in-tibet.jpg"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page.html"
        >
          Seven Years in Tibet
        </a>
      </h3>
    </article>
    <article
      className="small-movie-card catalog__movies-card"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <div
          className=""
        >
          <video
            autoPlay={false}
            className="player__video"
            poster="img/snatch.jpg"
            style={
              Object {
                "height": 175,
              }
            }
          />
        </div>
        <img
          alt="Snatch"
          height="175"
          src="img/snatch.jpg"
          width="280"
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="movie-page.html"
        >
          Snatch
        </a>
      </h3>
    </article>
  </div>
</section>
`;
